src/main/scala/tech/cryptonomic/nautilus/cloud/application/ApiKeyApplication.scala

```diff
@@ -45,13 +52,7 @@ class ApiKeyApplication[F[_]: Monad](conseilConfig: ConseilConfig, apiKeyService
 
   /** Returns all API keys from the DB */
   def getApiKeysForEnv(apiKey: String, environment: Environment): F[Permission[List[String]]] =
-    Either
-      .cond(
-        conseilConfig.keys.contains(apiKey),
-        apiKeyService.getAllApiKeysForEnv(environment),
-        AccessDenied("Wrong API key").pure[F]
-      )
-      .bisequence
+    requiredApiKey(apiKey, conseilConfig.keys)(apiKeyService.getAllApiKeysForEnv(environment))
```

Comment:
Could you help me clarify: what's in conseilConfig.keys?
Is that a global key set, a specific credential-enabled set, a specific user-set?

Reply:
I don't understand what it is that it's being verified by the requiredApiKey call, in the end.

Reply:
`conseilConfig.keys` are apiKeys used to validate http calls between services. Some of endpoints can work with validation by apiKeys and session, and some only with validation by session.